build.rs

```diff
@@ -1,11 +1,57 @@
 // Copyright The ocicrypt Authors.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::fs::File;
+use std::io::{Read, Write};
+use ttrpc_codegen::Customize;
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let protos = vec!["src/utils/proto/keyprovider.proto"];
     #[cfg(feature = "keywrap-keyprovider")]
     tonic_build::configure()
         .build_server(true)
-        .out_dir("src/utils/")
-        .compile(&["src/utils/proto/keyprovider.proto"], &["src/utils"])?;
+        .out_dir("src/utils/grpc")
+        .compile(&protos, &["src/utils"])?;
+
+    ttrpc_codegen::Codegen::new()
+        .out_dir("src/utils/ttrpc")
+        .inputs(&protos)
+        .include("src/utils")
+        .rust_protobuf()
+        .customize(Customize {
+            async_all: true,
+            ..Default::default()
+        })
+        .run()
+        .expect("Gen code failed.");
+
+    // To fix clippy warnings of code generated from ttrpc_codegen
```

Comment:
Can we use `#[rustfmt::skip]` to skip the fmt check for generated `keyprovider_ttrpc` module?

Reply:
No these are not rustfmt errors, these are something else, you can check here https://github.com/containerd/ttrpc-rust/blob/master/example/build.rs